frontend/src/metabase/components/Select.jsx

```diff
@@ -26,10 +37,14 @@ export default class Select extends Component {
     }
 
     render() {
-        var selectedName = this.props.value ? this.props.optionNameFn(this.props.value) : this.props.placeholder;
+        const value = this.props.changeOnSelect ?
```

Comment:
Yeah you're definitely right about this. Didn't put enough thought into this one.